litellm/tests/test_caching.py

```diff
@@ -1569,3 +1569,102 @@ async def test_redis_semantic_cache_acompletion():
     )
     print(f"response2: {response2}")
     assert response1.id == response2.id
+
+
+@pytest.mark.skip(reason="beta test - new dual semantic cache")
+def test_dual_semantic_cache_completion():
+    litellm.set_verbose = True
+    import logging
+
+    logging.basicConfig(level=logging.DEBUG)
+
+    random_number = random.randint(
+        1, 100000
+    )  # add a random number to ensure it's always adding /reading from cache
+
+    print("testing semantic caching")
+    litellm.cache = Cache(
+        type="dual-semantic",
+        host=os.environ["REDIS_HOST"],
+        port=os.environ["REDIS_PORT"],
+        password=os.environ["REDIS_PASSWORD"],
+        similarity_threshold=0.8,
+        redis_semantic_cache_embedding_model="local/multi-qa-MiniLM-L6-cos-v1",
```

Comment:
so do i need to run a model locally for this to work?

Reply:
it's a super tiny embedding model using sentence transformers, so yes, but the model is run by the cache itself